modules/shared/src/main/scala/org/tessellation/security/signature/Signed.scala

```diff
@@ -99,17 +98,19 @@ object Signed {
     def isSignedExclusivelyBy(signers: Set[Id]): Boolean =
       signed.proofs.map(_.id).toSortedSet.unsorted === signers
 
-    def hasValidSignature[F[_]: Async: SecurityProvider: KryoSerializer]: F[Boolean] =
+    def hasValidSignature[F[_]: Async: SecurityProvider: Hasher](implicit encoder: Encoder[A]): F[Boolean] =
       validProofs(None).map(_.isRight)
 
-    def hasValidSignature[F[_]: Async: SecurityProvider: KryoSerializer](toBytes: A => F[Array[Byte]]): F[Boolean] =
+    def hasValidSignature[F[_]: Async: SecurityProvider: Hasher](toBytes: A => F[Array[Byte]]): F[Boolean] = {
+      implicit val encoder: Encoder[A] = Encoder.instance(_ => Json.Null)
```

Comment:
Can you explain what is this about?

Reply:
Yep - I have two cases:
1. check `hasValidSignature` using the default serializer logic (kryo in our case)
2. check `hasValidSignature` using the custom serializer logic (provided fn `toBytes: A => F[Array[Byte]]`

Both calls `validProofs` and because I didn't want to write two `validProofs` methods, one working on kryo and second one working on custom provided fn, I pass the optional argument (default None for default logic). BUT I need to pass an encoder as well for the default logic. So for the second case, I pass the "noop" encoder. 

As I think about this right now, I can require `validProofs` to expect either the encoder (explicitly) or the custom logic fn. Should be better approach. Changing now!

Reply:
changed https://github.com/Constellation-Labs/tessellation/compare/93eb312acde435334d89f35406ecc5f56f59b4b5..5ecc90387aba4e8634bd69e513152ad2576ae76b